packages/core/src/Popout/Popout.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { RefObject, useCallback, useEffect, useLayoutEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import { Box } from '..'
+
+export interface IPopoutProps {
+  trigger: JSX.Element
+  content?: JSX.Element | undefined
+  triggerRef?: RefObject<HTMLElement>
+  onOpen?: () => void
+  onClose?: () => void
+}
+
+const TriggerWrapper = styled(Box)`
+  position: relative;
+  width: fit-content;
+`
+
+const OnClickHandler = styled(Box)`
+  position: absolute;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  cursor: pointer;
+`
+
+const Shadow = styled(Box)`
+  position: fixed;
+  top: 0;
+  left: 0;
+  cursor: pointer;
+  width: 100vw;
+  height: 100vh;
+  z-index: 1;
+  transition: opacity 250ms ease-in-out;
+`
+
+const PopoutModal = styled(Box)`
+  position: absolute;
+  z-index: 2;
+  background-color: white;
+  transition: padding 200ms ease-in-out, margin 200ms ease-in-out, border-radius 200ms ease-in-out;
+`
+
+export const Popout = (props: IPopoutProps) => {
+  const { trigger, content, onOpen, onClose, triggerRef } = props
+  const [isOpen, setIsOpen] = useState(false)
+  const [height, setHeight] = useState(0)
+  const [padding, setPadding] = useState(0)
+  const [opacity, setOpacity] = useState(0)
+  const baseInputRef = useRef<HTMLDivElement>(null)
+  const modalRef = useRef<HTMLDivElement>(null)
+
+  const handleOpen = useCallback(() => {
+    setIsOpen(true)
+    setTimeout(() => {
+      setPadding(16)
+      setOpacity(0.5)
+      /* istanbul ignore next */
+      triggerRef?.current?.focus()
+      /* istanbul ignore next */
+      onOpen?.()
+    })
+  }, [triggerRef, onOpen])
+
+  const handleClose = useCallback(() => {
+    setIsOpen(false)
+    setTimeout(() => {
+      setPadding(0)
+      setOpacity(0)
+      /* istanbul ignore next */
+      onClose?.()
+    })
+  }, [onClose])
+
+  useLayoutEffect(() => {
+    setHeight(baseInputRef.current.clientHeight)
+  }, [trigger])
+
+  useEffect(() => {
+    const escapeListener = (e: KeyboardEvent) => {
+      if (isOpen && e.key === 'Escape') handleClose()
+    }
```

Comment:
I noticed that tabbing out of the focused input has some weird behavior. Dismissing via the Escape key works as expected.

https://user-images.githubusercontent.com/3260642/205312036-63d854a2-b75b-4b50-95a6-bff3114035eb.mp4